src/components/ec/cuda/ec_cuda.c

```diff
@@ -319,6 +325,18 @@ static ucc_status_t ucc_ec_cuda_init(const ucc_ec_params_t *ec_params)
             return UCC_ERR_NOT_SUPPORTED;
         }
     }
+
+    if (cfg->use_cooperative_launch == 1) {
+        cudaDeviceGetAttribute(&supportsCoopLaunch,
+                               cudaDevAttrCooperativeLaunch, device);
+        if (CUDA_SUPPORTS_COOP_LAUNCH == 0 || supportsCoopLaunch == 0) {
+            ec_error(&ucc_ec_cuda.super,
+                     "CUDA cooperative groups are not supported. "
+                     "Please set UCC_EC_CUDA_USE_COOPERATIVE_LAUNCH to 0");
+            return UCC_ERR_NOT_SUPPORTED;
```

Comment:
maybe print warning but continue and use noncooperative launch?

Reply:
I fixed that in the last commit. Now, the parameter `use_cooperative_launch` is initialized to 1 by default, and if the device does not support coop launch it is switched to 0 and a debug message is printed with ec_debug.